src/infra/ChatAnalyticsReporter.ts

```diff
@@ -17,12 +18,19 @@ export class ChatAnalyticsReporter {
    * when the endpoint field from config is not provided.
    */
   private readonly endpoint: string;
+  /** Whether to enable session tracking for analytics events */
+  private readonly sessionTrackingEnabled: boolean;
 
   constructor(
-    { apiKey, env, region, endpoint }: ChatAnalyticsConfig,
+    { apiKey, env, region = 'US', endpoint, sessionTrackingEnabled }: ChatAnalyticsConfig,
     private httpRequesterService: HttpRequesterService
   ) {
     this.apiKey = apiKey;
+    if (sessionTrackingEnabled === undefined) {
```

Comment:
do you want to handle null here?

Reply:
from typescript perspective, it should only be `boolean | undefined` unless someone does something weird or or use js only. I can add the check for it and have it set to default, no harm.

Reply:
```
function foo (region = 'US', sessionTrackingEnabled = region === 'US' ? true : false) {
   console.log(region, sessionTrackingEnabled);
}
foo()
// outputs: US true 
foo('EU')
// outputs: EU false
```

Reply:
updated